src/libraries/System.Runtime.InteropServices.RuntimeInformation/tests/PlatformVersionTests.cs

```diff
@@ -0,0 +1,238 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Xunit;
+
+namespace System.Runtime.InteropServices.RuntimeInformationTests
+{
+    public class PlatformVersionTests
+    {
+        public static IEnumerable<object[]> AllKnownOsPlatforms()
+        {
+            yield return new object[] { OSPlatform.Windows };
+            yield return new object[] { OSPlatform.Linux };
+            yield return new object[] { OSPlatform.OSX };
+            yield return new object[] { OSPlatform.Browser };
+        }
+
+        [Fact]
+        public void IsOSPlatformOrLater_Null_ThrowsArgumentNullExceptionWithArgumentName()
+        {
+            Assert.Throws<ArgumentNullException>("platformName", () => RuntimeInformation.IsOSPlatformOrLater(null));
+        }
+
+        [Fact]
+        public void IsOSPlatformOrLater_Empty_ThrowsArgumentNullExceptionWithArgumentName()
+        {
+            Assert.Throws<ArgumentException>("platformName", () => RuntimeInformation.IsOSPlatformOrLater(string.Empty));
+        }
+
+        [Theory]
+        [InlineData("ios")] // missing version number
+        [InlineData("ios14")] // ios14.0 is fine, ios14 is not
```

Comment:
I missed this detail during the review. Why is `ios14` disallowed? /cc @terrajobst

Reply:
I wanted to make it consistent with the `Version.TryParse` behavior which does not support just major versions like "14" ("14.0" is fine)

should this be changed? (it's not a problem)

Reply:
I would expect folks to get annoyed having to put `.0` on the versions, but I'll defer to @terrajobst if there's a reason to not support it that I can't think of.

Reply:
I talked with @terrajobst on this and the reason we want to prevent "ios14" from working is that with some of our other parsing of version numbers, this becomes ambiguous between "14.0" and "1.4" where the "." is inferred (like with "net50").

Reply:
thanks for the clarification. I added xml docs and exception message that is very clear about these requirements.